articles/101_build_tool.md

```diff
@@ -48,43 +54,74 @@ In the ROS ecosystem [rosdep](http://wiki.ros.org/rosdep) can be used for this.
-It determines the dependency graph and invokes the package specific build tool of each package in topological order.
-The build tool itself should know as little as possible about the build system used for a specific package.
-Just enough in order to know how to setup the environment for it, invoke the build, and setup the environment to use the built package.
+A build tool defines a manifest format for packages to follow to define the invocation of the respective build tool, and the dependencies to other packages.
 
-The build system on the other hand operates on a single package.
-In the case a package uses e.g. `CMake` the responsibility of the tool comes down to invoke the common steps `cmake`, `make`, `make install` for this package.
-As another example for a package using `Autotools` the steps could look like `configure`, `make`, `make install`.
+### Development Environment Setup
 
-### Environment Setup
+Invoking a buildsystem for a package implies also setting up environment variables before the process.
+Examples are the `CMAKE_PREFIX_PATH` and the `LD_LIBRARY_PATH`.
+For consistency of the build result, those variables should be restricted, allowing the build to only access declared dependencies.
 
-A very important part beside the actual build of a package is the environment setup.
-For a CMake package e.g. a dependency discovered using the CMake function `find_package` needs to be either in a common location or the location must be provided through the environment variable `CMAKE_PREFIX_PATH`.
-On the other hand after a package has been built and installed the environment might need to be extended to use the package.
+For various reasons, it is beneficial to also allow developers to easily manually invoke the buildsystem for one package.
+This requires the buildsystem to provide the environment setup by itself.
+
+### Subtask invocation
+
+Build systems define various tasks, such as testing, compiling, generating documentation, installing.
+A build tool must provide it's own abstract set of tasks to be mapped to the tasks of different build systems.
+
+### Isolated installation
+
+For the ROS ecosystem it is important that packages can be installed to any target location in a filesystem.
+
+After a package has been built and installed to a target location, the environment might need to be extended to use the package.
 E.g. when a package installs a shared library in a non-default location the environment variable `LD_LIBRARY_PATH` (or `PATH` on Windows) needs to be extended to include the containing folder.
 
-The functionality to setup these environment variables can be provided by either the build tool or the build system.
-In the latter case the build tool only needs to know how the build system exposes the environment setup in order to reuse it.
+The build tool make it easy to extend the environment to run code from installed packages, or to build other packages on top of installed ones.
```

Comment:
So it would be a "nice-to-have" feature to extract the responsibility for the environment from the build-tool? If so it should be kept, could be marked more explicitly as a possible goal.
I removed it because it looked like a technical remark, not something related to a goal.